### ISSUE TITLE
'-' within schema causes invalid stuct name to be created '-' can't be within struct name

### ISSUE THREAD

**Issue by nlacey** [fiorix/wsdl2go#47]
injested wsdl files with - in there field name, 
In order to get the program to create the go code,
I needed to go through encoder.go and add string.Replace on '-' with "" one the various tag fields

**Comment by fiorix** [comment]
Can you give an example?

**Comment by nlacey** [comment]
I'm sorry for how long this took me.
If you replace nick-name  with nickname, in the included file, the problem will work.

[problem_example_wsdl.txt](https://github.com/fiorix/wsdl2go/files/1580137/problem_example_wsdl.txt)

I messed around wtih wsdlgo/encoder.go so struct names wouldn't include the '-' character, which allowed your code to run.

**Comment by fiorix** [comment]
Alright, so this is mainly about removing `-` from function names and variables but keeping them in the XML tags? PR?

**Comment by nlacey** [comment]
correct,  attached export.go that worked with example file, 
[encoder.txt](https://github.com/fiorix/wsdl2go/files/1587436/encoder.txt)

created cleanString function (currently removes '-' and ' ')
Not sure how complete it is, but it was able to convert the previously attached file thanks!

**Comment by fiorix** [comment]
Yeah we need to fix this. @flavioayra just reported the same problem, finding "-" and "." in type/op names.

**Comment by fiorix** [comment]
Should be good now.

**Comment by warroyo** [comment]
@fiorix i am still seeing this issue with "." in the name, was this added to fix "." or just "-"

**Comment by fiorix** [comment]
I think I missed the "." in the regex that makes it a valid Go symbol: https://github.com/fiorix/wsdl2go/blob/88777812a5daba043066aae0e7dd72b0a6343f98/wsdlgo/encoder.go#L1243

**Comment by warroyo** [comment]
@fiorix in this case the "." is actually invalid in the type name , would adding that to the `validGoSymbol` cause it to be replaced?

**Comment by fiorix** [comment]
PR?

**Comment by prologic** [comment]
@fiorix I'm also seeing this as well.

For example:

```console
$ wsdl2go < ParseAddress.wsdsl > ParseAddr.go
2022/10/03 12:04:06 generated bad code: 3:13: expected ';', found '-'
    1	// Code generated by wsdl2go. DO NOT EDIT.
    2
    3	package plsp-public
...
```

This WSDL comes from: https://information.qld.gov.au/service/Addressing/ValidationService/PLSplusPublic/soap?wsdl

**Comment by prologic** [comment]
#47 fixes a bunch of bugs I found with generating type names, method names and so on. This resolves my problem as well as the one found in this issue. I'm not happy with the code, but it generated valid Go code finally 😅

**Comment by prologic** [comment]
FWW I achieved much better results by using the `wsdlgen` cli utility found in this package:

- https://github.com/droyo/go-xml

**Comment by fiorix** [comment]
Fixed: '-' characters in schema names are now stripped from generated Go identifiers and package names.
